### PR TITLE
Datastore API

### LINKED ISSUE #83: API for Datastore
We need an API implementation to do CRUD operations on the `datastores` object.

- [x] Listing of datastores (GET)
- [x] Creation of a datastore (POST)
- [x] Viewing of individual datastores (GET)
- [x] Updating individual fields of a field (PUT)
- [x] Deletion of an object (DELETE)

##### GET: `https://minsql/api/v1/datastores/`:

- Should return a list of all datastores
- Pagination support is needed
- Total number of datastores should be returned with each call

##### POST: `https://minsql/api/v1/datastores/`:

- Creates a `datastore` object

##### GET: `https://minsql/api/v1/datastores/<datastore_NAME>/`

- Should full details for an individual datastore 
- Listing of secrets is not allowed (SECRET KEY)
##### PUT: `https://minsql/api/v1/datastores/<datastore_NAME>/`

- Should allow update of the fields on the datastore
##### DELETE: `https://minsql/api/v1/datastores/<datastore_NAME>/`

- Should remove the datastore object

### PR DESCRIPTION
**Pull request by dvaldivia**
fixes #83 
This PR depends on #102

### CODE REVIEW

File: src/api/api_datastores.rs [hunk 1]
```diff
@@ -0,0 +1,307 @@
+// This file is part of MinSQL
```
Comment (donatello) [inline src/api/api_datastores.rs]: Is it necessary to prefix the filename with `api_`? Let's not use unnecessary conventions.

**Review by donatello — CHANGES_REQUESTED** [review]
Early round, still reviewing.

File: src/api/api_datastores.rs (hunk 1 above)
Comment (dvaldivia) [reply src/api/api_datastores.rs]: since I have other 2 PRs that rely on this file, let me fix this on a separate PR after #108 is merged

**Comment by dvaldivia** [comment]
#105 merged this code

**Pull request closed without merging**